pkg/k8s/controllers/reconciler.go

```diff
@@ -0,0 +1,276 @@
+// Copyright 2021 Outreach Corporation. All Rights Reserved.
+
+// Description: This file defines a kubernetes controller for v1/PostgresqlDevenvDatabase.
+package controllers
+
+import (
+	"context"
+	"time"
+
+	"github.com/getoutreach/gobox/pkg/k8s/resources"
+	"github.com/sirupsen/logrus"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// requeueIntervals are the intervals with which we will requeue failed CRs
+// last value (also available as maxRequeueInterval) is used after we tried all
+var requeueIntervals = []time.Duration{
```

Comment:
These seem restrictive, what if a controller wants to retry in 100ms? is it possible to override these?

Reply:
yes - controller can ask for a different requeue - and Reconciler will honor it. I added code for this - see endReconcile.